pkg/kubelet/dockertools/manager.go

```diff
@@ -375,7 +375,10 @@ func (dm *DockerManager) GetPodStatus(pod *api.Pod) (*api.PodStatus, error) {
 	for _, status := range pod.Status.ContainerStatuses {
 		oldStatuses[status.Name] = status
 		if status.LastTerminationState.Terminated != nil {
-			lastObservedTime[status.Name] = status.LastTerminationState.Terminated.FinishedAt
+			timestamp, ok := lastObservedTime[status.Name]
+			if !ok || (ok && timestamp.Before(status.LastTerminationState.Terminated.FinishedAt)) {
```

Comment:
The bug was there for a while, but was masked by the previous restart count bug, which was fixed by my last PR. We are now seeing really high restart counts due to incorrect accounting.